x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateDiff.java

```diff
@@ -114,12 +114,8 @@ private static int safeInt(long diff) {
         }
 
         private static long diffInMinutes(ZonedDateTime start, ZonedDateTime end) {
-            long secondsDiff = diffInSeconds(start, end);
-            if (secondsDiff > 0) {
-                return (long) Math.ceil(secondsDiff / 60.0d);
-            } else {
-                return (long) Math.floor(secondsDiff / 60.0d);
-            }
+            // Truncate first to minutes (ignore any seconds and sub-seconds fields)
+            return (end.toEpochSecond() / 60) - (start.toEpochSecond() / 60);
```

Comment:
Isn't this equivalent to `(end.toEpochSecond() - start.toEpochSecond()) / 60`?

Reply:
No, it's not the same. We need to truncate each date first, if we do a final truncation on the result of the subtraction it fails for example for:

```SELECT DATEDIFF(mi, '2019-12-31T20:22:33.987654321', '2022-01-01T04:33:22.123456789');
```
will return `1053130` instead of `1053131`

and this is tested here: https://github.com/elastic/elasticsearch/pull/49252/files#diff-7c21301d54284bcf7b693c9f1eb92ba8R178

Reply:
Sounds good.